Set default project path where a project was last created

KoBeWi opened the pull request:
Closes https://github.com/godotengine/godot-proposals/issues/1312

KoBeWi committed:
Set default project path where a project was last created

akien-mga commented:
See https://github.com/godotengine/godot-proposals/issues/1312#issuecomment-2506275369, I don't think this is the right approach.

KoBeWi commented:
I added a checkbox:
![image](https://github.com/user-attachments/assets/e1f9cf5f-32ee-4055-865c-92819eceee77)
Though it needs a better place.

Changed default path to empty string. The checkbox will be automatically enabled if the path is empty.

KoBeWi commented:
I'm not sure about removing the checkbox, although maybe it's less useful if the setting is not exposed 🤔
